vsintegration/src/FSharp.Editor/DocumentDiagnosticAnalyzer.fs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Composition
+open System.Collections.Immutable
+open System.Threading
+open System.Threading.Tasks
+
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.Diagnostics
+open Microsoft.CodeAnalysis.Host.Mef
+open Microsoft.CodeAnalysis.Text
+open Microsoft.CodeAnalysis.SolutionCrawler
+
+open Microsoft.FSharp.Compiler
+open Microsoft.FSharp.Compiler.SourceCodeServices
+open Microsoft.FSharp.Compiler.Range
+
+open Microsoft.VisualStudio.FSharp.LanguageService
+
+[<DiagnosticAnalyzer(FSharpCommonConstants.FSharpLanguageName)>]
+type internal FSharpDocumentDiagnosticAnalyzer() =
+    inherit DocumentDiagnosticAnalyzer()
+
+    static member ConvertError(filePath: string, sourceText: SourceText, error: FSharpErrorInfo) =
+        let id = "FS" + error.ErrorNumber.ToString()
```

Comment:
This might result in funny looking error codes.  E.g., error number 42 would be formatted as "`FS42`" where the command line compiler would format it as "`FS0042`".  Changing this to `.ToString("0000")` should fix it.